### ISSUE TITLE
Regarding the issue of missed frames

### ISSUE THREAD

**Issue by 408550969** [ubicomplab/rPPG-Toolbox#265]
Hi,I have a video with a frame rate of 30 frames per second, but it experiences frame dropping. Assuming it drops one frame every 30 frames, should I set FS to 29 or FS to 30 in the training config?

**Comment by 408550969** [comment]
Thanks, I have another question. Will the delay between the image and label cause the model to fail to converge? I collected a video and found that as long as the difference between the image (which has already calculated the camera delay) and the label exceeds plus or minus 133ms, the model cannot converge. Is RPPG very sensitive to latency?

**Comment by yahskapar** [comment]
Hi @408550969,

There's definitely a possibility that some combinations of models and datasets are more sensitive to synchronization error, which in turn could lead to a failure to converge. Can you share more details (e.g., how you identified the model being unable to converge, such as plots of the training and loss curves which can be produced by this toolbox)?

Usually this sensitivity has more to do with the loss function than a specific model from what I understand. If correcting the synchronization error is challenging, you could try using loss functions such as the Maximum Cross-Correlation (MCC) as suggested by [this paper](https://arxiv.org/pdf/2402.11913v1.pdf).

**Comment by 408550969** [comment]
I identified by testing the MAE of both the test set and the training set that when the delay is large, not only does the MAE of the test set reach tens, but the MAE of the training set also reaches tens.
Thanks, I will consider using MCC as the loss.

**Comment by 408550969** [comment]
I have another question, does video encoding have a significant impact on accuracy? For example, if I use lossy compression, what is the difference in MAE compared to lossless images?

**Comment by yahskapar** [comment]
Hi @408550969,

I recommend reading section 10.5 (titled 'Video Compression') of this [excellent review article](https://dl.acm.org/doi/pdf/10.1145/3558518), as well as any of the cited works in that section that sound interesting to you and relate to the effects of video compression on the task of rPPG. To put it briefly and based on my understanding, yes, there is a difference and I'd expect compression that has more temporal effects to subsequently have a greater effect on making your SNR and possibly your MAE worse.

**Comment by 408550969** [comment]
Thanks!
